api/base/parsers.py

```diff
@@ -15,6 +17,29 @@ class JSONAPIParser(JSONParser):
     media_type = 'application/vnd.api+json'
     renderer_class = JSONAPIRenderer
 
+    def flatten_relationships(self, relationships):
+        """
+        Flattens relationships dictionary which has information needed to create related resource objects.
+
+        Validates that formatting of relationships dictionary is correct.
+        """
+        if not isinstance(relationships, dict):
+            raise ParseError()
+        related_resource = relationships.keys()[0]
```

Comment:
Moving the larger fix to the backlog has always been an option, but absolutely leave a source code comment with a note to future developers as part of this PR. (This is a known edge case that *will* break their stuff, and *will* be hard to track down once we've forgotten about it)